crates/crypto/bn254/src/utils.rs

```diff
@@ -15,16 +14,13 @@ use rust_bls_bn254::pairing;
 pub fn map_to_curve(digest: &[u8]) -> G1Affine {
     let one = Fq::one();
     let three = Fq::from(3u64);
-    let big_int = BigUint::from_bytes_be(digest);
     let mut bytes = [0u8; 32];
-    big_int
-        .to_bytes_be()
-        .iter()
-        .rev()
-        .enumerate()
-        .for_each(|(i, &b)| bytes[i] = b);
 
-    let mut x = Fq::from_le_bytes_mod_order(&bytes);
+    // This adds padding to the digest to make it 32 bytes. Go SDK automatically does it.
+    // This makes HashToCurve same for both Rust and Go SDK.
+    bytes[..digest.len()].copy_from_slice(digest);
```

Comment:
this panics if lenght of digest is larger than 32

Reply:
ahh I can put the check

Reply:
@pablodeymo not rust expert but do we want to throw error here or just copy the first 32 bytes? or is there a way to force digest to be 32 byte?

Reply:
can't we just do this? `pub fn map_to_curve(digest: &[u8; 32]) -> G1Affine {`